lib/rbi/rbs_printer.rb

```diff
@@ -0,0 +1,1013 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  class RBSPrinter < Visitor
+    class Error < RBI::Error; end
+
+    sig { returns(T::Boolean) }
+    attr_accessor :print_locs, :in_visibility_group
+
+    sig { returns(T.nilable(Node)) }
+    attr_reader :previous_node
+
+    sig { returns(Integer) }
+    attr_reader :current_indent
+
+    sig { params(out: T.any(IO, StringIO), indent: Integer, print_locs: T::Boolean).void }
+    def initialize(out: $stdout, indent: 0, print_locs: false)
+      super()
+      @out = out
+      @current_indent = indent
+      @print_locs = print_locs
+      @in_visibility_group = T.let(false, T::Boolean)
+      @previous_node = T.let(nil, T.nilable(Node))
+    end
+
+    # Printing
+
+    sig { void }
+    def indent
+      @current_indent += 2
+    end
+
+    sig { void }
+    def dedent
+      @current_indent -= 2
+    end
```

Comment:
Since these should always be called in matching pairs, what do you think about making a block-based API?

```suggestion
sig do
  type_parameters(:R)
    .params(block: T.proc.returns(T.type_parameter(:R))
    .returns(T.type_parameter(:R))
end
def indent(&block)
  @current_indent += 2
  yield
ensure
  @current_indent -= 2
end
```

Called with e.g.

```ruby
indent do
  visit_all(node.nodes)
end
```